src/node.h

```diff
@@ -148,6 +148,77 @@ typedef intptr_t ssize_t;
 
 namespace node {
 
+  enum NodeInstanceType { MAIN, WORKER };
```

Comment:
Style: don't indent.  Also, I'd like to see this moved to e.g. node_internals.h.  node.h is a public header.  Adding new APIs should not be done lightly.